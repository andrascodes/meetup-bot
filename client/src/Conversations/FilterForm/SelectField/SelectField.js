import React from 'react'

export const SelectField = props => {
  
  const options = props.options.map(opt => <option value={opt.value} key={opt.value}>{opt.text}</option> )

  return (
    <div className="SelectField">
      <label htmlFor={props.name} className="SelectLabel">{props.label}</label>
      <select 
        name={props.name} 
        id={props.name}
        onChange={props.handleChange}
        defaultValue={props.defaultValue}
      >
        {options}
      </select>
    </div>
  )
}

SelectField.PropTypes = {
  options: React.PropTypes.arrayOf(React.PropTypes.shape({
    value: React.PropTypes.number,
    text: React.PropTypes.string
  })).isRequired,
  name: React.PropTypes.string.isRequired,
  label: React.PropTypes.string.isRequired,
  handleChange: React.PropTypes.func.isRequired,
  defaultValue: React.PropTypes.string
}