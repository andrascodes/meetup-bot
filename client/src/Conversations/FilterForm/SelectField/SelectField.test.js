import React from 'react'
import renderer from 'react-test-renderer'

import { SelectField } from './SelectField'

describe('SelectField', () => {

  test('renders correctly', () => {
    
    const tree = renderer.create(
      <SelectField 
        options={[
          {
            value: 'label1 label2',
            text: 'label1 label2'
          }, 
          {
            value: 'label3 label4',
            text: 'label3 label4'
          }
        ]}
        name={'selectfield'}
        label={'Label'}
        handleChange={() => {}}
        defaultValue={'null'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('renders correctly when no defaultValue is set', () => {
    
    const tree = renderer.create(
      <SelectField 
        options={[
          {
            value: 'label1 label2',
            text: 'label1 label2'
          }, 
          {
            value: 'label3 label4',
            text: 'label3 label4'
          }
        ]}
        name={'selectfield'}
        label={'Label'}
        handleChange={() => {}}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('throws PropType error', () => {
    const render = () => renderer.create(
      <SelectField
        handleChange={() => {}}
      />
    )
    expect(render).toThrowError(`The prop 'options' has to be defined, pass an empty array ([])`)
  })

})