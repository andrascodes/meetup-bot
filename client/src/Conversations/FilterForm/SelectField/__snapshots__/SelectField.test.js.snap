exports[`SelectField renders correctly 1`] = `
<div
  className="SelectField">
  <label
    className="SelectLabel"
    htmlFor="selectfield">
    Label
  </label>
  <select
    defaultValue="null"
    id="selectfield"
    name="selectfield"
    onChange={[Function]}>
    <option
      value="label1 label2">
      label1 label2
    </option>
    <option
      value="label3 label4">
      label3 label4
    </option>
  </select>
</div>
`;

exports[`SelectField renders correctly when no defaultValue is set 1`] = `
<div
  className="SelectField">
  <label
    className="SelectLabel"
    htmlFor="selectfield">
    Label
  </label>
  <select
    defaultValue={undefined}
    id="selectfield"
    name="selectfield"
    onChange={[Function]}>
    <option
      value="label1 label2">
      label1 label2
    </option>
    <option
      value="label3 label4">
      label3 label4
    </option>
  </select>
</div>
`;
